### DISCUSSION TITLE
Hacktoberfest 2023

### DISCUSSION THREAD

**Discussion by CeciliaHinga** [ushahidi/platform#4650] [Hacktoberfest]
Welcome to Ushahidi and the **Hacktoberfest** month!

We are so excited you are here!! We'd love to celebrate Open Source with you in October 2023 🌻 🎉 💐

# 

We have compiled a list of helpful resources and steps to help you make successful Hacktoberfest contributions. Ensure to not skip any of the steps below so that your submissions/pull requests does not get disqualified by the hacktoberfest team.

- **Step 1 -** Check out the [hacktoberfest website](https://hacktoberfest.com) for details about how hacktoberfest works, what kind of submissions/pull requests are acceptable etc. Don't forget to signup on their website so that your submissions can count.
- **Step 2 -** Get familiar with the Ushahidi community:
    - Read through our [Code of Conduct](https://docs.ushahidi.com/platform-developer-documentation/code-of-conduct)
    - Join and say hello on our [community Gitter channel](https://app.element.io/#/room/#ushahidi_Community:gitter.im) to communicate, help and get help from the community.
- **Step 3 -** Get your developer environment setup (If you get stuck, reach out to the community Gitter channel to get help):
    - Checkout the instructions to setup and launch our [platform client (mzima)](https://github.com/ushahidi/platform-client-mzima#platform-client-mzima) locally.
    - Checkout the instructions to setup and launch our [backend platform API](https://github.com/ushahidi/platform#setup-essentials) locally.
- **Step 4 -** Explore the Ushahidi Platform and the different concepts like surveys (also called "forms" in some places) and posts. You can explore by studying and clicking around on the platform's user interface, or through our [user documentation](https://docs.ushahidi.com/ushahidi-platform-user-manual) (**Note:** that the user documentation still uses the legacy/old platform client in its examples but the functionalities are largely the same. You can get clarification about any confusing thing on the community Gitter channel)

# 

**Share your successful setup:** Once you have completed steps 1-4, send a reply to this discussion with screenshots of your setup in the browser and select an issue to work with.

**Finding issues tickets to work on and working on issues:** You can find issues to work on through our [OSS community board](https://github.com/orgs/ushahidi/projects/16/views/1?visibleFields=%5B%22Title%22%2C%22Assignees%22%2C%22Status%22%2C57223039%2C57227750%2C%22Labels%22%2C%22Linked+pull+requests%22%2C%22Repository%22%5D). We have made it easy - check for issue tickets with the `hacktoberfest` label on them, you will also find the difficulty level on the issue tickets to help you easily identify those that you can tackle based on your experience and/or strengths. Take note of these when working on issues:
- You can only work on one issue at a time. We are looking for the quality of the contributions rather than the number of contributions when evaluating the applications.
- Once you are assigned to an issue, we want you to submit a pull request within 2 days, even if it is not yet complete or a working solution. This is to make our contributors work "in the open" as is often done in Open Source, and make sure no issues are not being worked on. If you don't show progress within 3 days, we will open it up for other participants. GitHub has a good guide on [how to submit pull requests from a fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork).
- Remember that Ushahidi is an open-source community. The mentors are trying their best to reply to your questions, but there may be some delays. By helping other contributors out in our community Gitter channel, you are also helping the mentors. This is recognized by us and seen as something very positive in evaluating your submissions. So if you see someone asking for help in the chat and know the answer, please help them out! Keep the tone nice and collaborative.

**Reporting issues:** Although for Hacktoberfest, submitting issue tickets is not counted as contributions. In open source, on github, and to us, they are contributions. We encourage you to report any (not yet reported) issues that you detect with any of our repositories or applications that you are working on. [Report issues for any of our repositories here](https://github.com/ushahidi/.github/issues/new). We will sort them out and send them to the appropriate columns on the OSS community board for you and other contributors to be able to start working on them.

**Privacy info/personal info:** During the application-process, Ushahidi will store your Github-handle in a google-sheet. This is done only to make the application process as smooth as possible and to keep track of which issues are being worked on and the progress of each issue. The information stored will be deleted once the application-process is over.

# 

Welcome once again, looking forward to the awesome contributions you will submit to our community 🌻 🎉 💐

**Comment by oyenuga17** [comment]
<img width="1672" alt="Screenshot 2023-10-10 at 13 04 06" src="https://github.com/ushahidi/platform/assets/64274826/235ae3e2-c52f-4da2-acf9-5817a99f1d65">
<img width="1669" alt="Screenshot 2023-10-10 at 13 04 41" src="https://github.com/ushahidi/platform/assets/64274826/0758b622-22a6-4e3b-8907-9bc79f0734d4">

-Ready to start 🎉

**Reply by Ifycode** [reply]
Welcome to Hacktoberfest at Ushahidi @oyenuga17  🎉

**Comment by shakirandagire** [comment]
Happy to begin Hacktoberfest contributions.
<img width="1512" alt="Screenshot 2023-10-19 at 23 01 28" src="https://github.com/ushahidi/platform/assets/6957960/e10e4462-55be-4f21-a5f6-5d64fc96c6b9">
<img width="1512" alt="Screenshot 2023-10-19 at 23 02 04" src="https://github.com/ushahidi/platform/assets/6957960/0f0a2076-dd52-4f80-84c3-28989eb3d4ce">

**Reply by Ifycode** [reply]
Welcome to Hacktoberfest at Ushahidi @shakirandagire   🎉

**Reply by shakirandagire** [reply]
Thank you @Ifycode

**Comment by MugoBrian** [comment]
Hello. I believe I am set to start Hacktoberfest contributions, I hope I am not late.
![platform-client-mzima](https://github.com/ushahidi/platform/assets/49617104/96ac240b-3e1a-4bae-becc-53cc19b8c418)
![platform-setup](https://github.com/ushahidi/platform/assets/49617104/abe2dc8e-1948-430c-acae-a9ffe45958ce)

**Reply by Ifycode** [reply]
No you are not too late. Welcome to Hacktoberfest at Ushahidi @MugoBrian 🎉
